src/edu/cmu/tartan/Game.java

```diff
@@ -248,7 +268,34 @@ public String getGameDescription(){
      */
     public boolean validate() {
         // TODO: This method is way too simple. A more thorough validation must be done!
-        return (gameName != null && gameDescription != null);
+		if(this.goals instanceof GameCollectGoal){
+			List<String> itemsListGoal = ((GameCollectGoal) this.goals).getItemsList();
+			int numItem = 0;
+
+			for (Room i : this.roomArrayList){
+				numItem += i.items.size();
```

Comment:
"+=" operation is not recommand.
It's looks good as below.

numItem = numItem + i.items.size();